### PR TITLE
Davidlee0602 browse api

### PR DESCRIPTION
**Pull request by davidlee0602**

### CODE REVIEW

File: client/components/Quizlist.jsx [hunk 1]
```diff
@@ -45,10 +42,12 @@ export const Quizlist = () => {
   return (
     <div className={classes.root}>
       <Box m={2}>
-        {quizzes.map((quiz, index) => (
+        {quizList.map((quiz) => (
           <Accordion expanded={expanded === quiz} onChange={handleChange(quiz)}>
```
Comment (dlimx) [inline client/components/Quizlist.jsx:46]: We should provide a `key` here for React optimization - `quiz.id` would be great

File: server/quizzes/controller.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { db } from '../database/firestore';
+
+// get quizzes from firestore
+export const getQuizzes = async (userQuery) => {
+  // search and category conditions
+  let quizSearch = '';
+  if (userQuery.search.length > 0 && userQuery.category.length < 1) {
+    quizSearch = db.collection('quizzes').where('name', '>=', userQuery.search).where('name', '<=', `${userQuery.search}~`);
+  } else if (userQuery.search.length > 0 && userQuery.category.length > 0) {
+    quizSearch = db.collection('quizzes').where('name', '>=', userQuery.search).where('name', '<=', `${userQuery.search}~`).where('category', '==', userQuery.category);
```
Comment (dlimx) [inline server/quizzes/controller.js:10]: Nice!

**Review by dlimx — APPROVED** [review]
Mostly LGTM - tiny comment

File: client/components/Quizlist.jsx (hunk 1 above)
Comment (davidlee0602) [reply client/components/Quizlist.jsx:46]: Thank you. I will make sure to add in the keys for optimization!

**Pull request merged**